lib/modules/manager/kustomize/extract.spec.ts

```diff
@@ -433,4 +433,259 @@ describe('modules/manager/kustomize/extract', () => {
+        host: 'https://itfs.mycompany.com/',
+        project: 'collection/project/_git/somerepos',
+        packageName:
+          'https://itfs.mycompany.com/collection/project/_git/somerepos',
+      },
+      {
+        url: 'git::https://itfs.mycompany.com/collection/project/_git/somerepos',
+        host: 'https://itfs.mycompany.com/',
+        project: 'collection/project/_git/somerepos',
+        packageName:
+          'https://itfs.mycompany.com/collection/project/_git/somerepos',
+      },
+      {
+        url: 'https://bitbucket.example.com/scm/project/repository.git',
+        host: 'https://bitbucket.example.com/',
+        project: 'scm/project/repository.git',
+        packageName: 'https://bitbucket.example.com/scm/project/repository.git',
+      },
+      {
+        url: 'ssh://git-codecommit.us-east-2.amazonaws.com/someorg/somerepo/somepath',
+        host: 'ssh://git-codecommit.us-east-2.amazonaws.com/',
+        project: 'someorg/somerepo',
+        packageName:
+          'ssh://git-codecommit.us-east-2.amazonaws.com/someorg/somerepo',
+      },
+      {
+        url: 'git@github.com/someorg/somerepo/somepath',
+        host: 'git@github.com:',
+        project: 'someorg/somerepo',
+      },
+      {
+        url: 'https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/',
+        host: 'https://github.com/',
+        project: 'kubernetes-sigs/kustomize',
+      },
+      {
+        url: 'ssh://git@github.com/kubernetes-sigs/kustomize//examples/multibases/dev',
+        host: 'git@github.com:',
+        project: 'kubernetes-sigs/kustomize',
+      },
+      {
+        url: 'https://example.org/path/to/repo//examples/multibases/dev',
+        host: 'https://example.org/',
+        project: 'path/to/repo',
+        packageName: 'https://example.org/path/to/repo',
+      },
+      {
+        url: 'https://example.org/path/to/repo.git/examples/multibases/dev',
+        host: 'https://example.org/',
+        project: 'path/to/repo.git',
+        packageName: 'https://example.org/path/to/repo.git',
+      },
+      {
+        url: 'ssh://alice@example.com/path/to/repo//examples/multibases/dev',
+        host: 'ssh://alice@example.com/',
+        project: 'path/to/repo',
+        packageName: 'ssh://alice@example.com/path/to/repo',
+      },
+      // {
+      //   url: 'https://authority/org/repo?ref=group/version',
+      //   host: 'https://authority/',
+      //   project: 'org/repo',
+      //   packageName: 'https://authority/org/repo',
+      // },
+      // {
+      //   url: 'https://authority/org/repo/?ref=includes_git/for_some_reason',
+      //   host: 'https://authority/',
+      //   project: 'org/repo',
+      //   packageName: 'https://authority/org/repo',
+      // },
+      // {
+      //   url: 'https://authority/org/repo/?ref=includes.git/for_some_reason',
+      //   host: 'https://authority/',
+      //   project: 'org/repo',
+      //   packageName: 'https://authority/org/repo',
+      // },
```

Comment:
The extraction might still fail for these URLs. Should I add a new regex to handle these or are we okay leaving these alone (no mainstream git-platform use these)